rasa/core/channels/telegram.py

```diff
@@ -59,38 +62,42 @@ async def send_text_with_buttons(
         :button_type reply: reply keyboard
         """
         if button_type == "inline":
-            button_list = [
-                [
-                    InlineKeyboardButton(s["title"], callback_data=s["payload"])
-                    for s in buttons
-                ]
-            ]
-            reply_markup = InlineKeyboardMarkup(button_list)
+            reply_markup = InlineKeyboardMarkup()
+            reply_markup.row_width = len(buttons)
+            for idx, button in enumerate(buttons):
+                reply_markup.add(
+                    InlineKeyboardButton(
+                        button["title"], callback_data=button["payload"]
+                    )
+                )
 
         elif button_type == "vertical":
-            button_list = [
-                [InlineKeyboardButton(s["title"], callback_data=s["payload"])]
-                for s in buttons
-            ]
-            reply_markup = InlineKeyboardMarkup(button_list)
+            reply_markup = ReplyKeyboardMarkup(
+                resize_keyboard=True, one_time_keyboard=True
+            )
+            for idx, button in enumerate(buttons):
+                if isinstance(button, list):
+                    reply_markup.add(KeyboardButton(s["title"]) for s in button)
```

Comment:
Why would `button` (in a buttons list) also be a list?

Reply:
fixed